### PR TITLE
RDM-4544 - filter upload file types

### PR DESCRIPTION
**Pull request by gbenadikar**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/RDM-4544

### Change description ###
Filter upload file types using field type regular expression

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/shared/components/palette/document/write-document-field.component.ts [hunk 1]
```diff
@@ -63,6 +64,9 @@ export class WriteDocumentFieldComponent extends AbstractFieldWriteComponent imp
         document.document_filename,
       );
     }
+    if (this.caseField.field_type) {
```
Comment (mario-paniccia) [inline src/shared/components/palette/document/write-document-field.component.ts]: is this check required? we always have a field type for a field isn't it?

File: src/shared/components/palette/document/write-document-field.component.ts [hunk 2]
```diff
@@ -63,6 +64,9 @@ export class WriteDocumentFieldComponent extends AbstractFieldWriteComponent imp
...
+      this.fileTypeFilter = this.caseField.field_type.regular_expression;
```
[the remark this answers is not in the document]
Comment (gbenadikar) [reply src/shared/components/palette/document/write-document-field.component.ts]: Had checked with Steve R during 3A, he said there won't be any requirement for restricting to filenames, only the file types matter.

File: src/shared/components/palette/document/write-document-field.component.ts (hunk 1 above)
Comment (gbenadikar) [reply src/shared/components/palette/document/write-document-field.component.ts]: refactored code

File: src/shared/components/palette/document/write-document-field.component.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (mario-paniccia) [reply src/shared/components/palette/document/write-document-field.component.ts]: 👍

**Pull request merged**